commons/src/main/scala/dagr/commons/reflect/ReflectionUtil.scala

```diff
@@ -236,13 +257,134 @@ object ReflectionUtil {
     else mirror.runtimeClass(typ.typeSymbol.asClass)
   }
 
-  /** Fetches the TypeSymbol (a Symbol representing a type, not a Type) for a given class. */
-  def classToTypeSymbol(clazz: Class[_]): TypeSymbol = {
-    mirror.classSymbol(clazz)
-  }
-
   /** Gets the annotation of type `T` of the class `clazz` */
   def findJavaAnnotation[T <: java.lang.annotation.Annotation](clazz: Class[_], annotationClazz: Class[T]): Option[T] = {
     Option(clazz.getAnnotation(annotationClazz))
   }
+
+  /**
+    * Constructs one or more objects from Strings.
+    *
+    * @param resultType the type that must be assignable to, from the return of this function (may be a collection or a single-valued type)
+    * @param unitType either the same as resultType or when resultType is a collection, the type of elements in the collection
+    * @param value one or more strings from which to construct objects.  More than one string should only be given for collections.
+    * @return an object constructed from the string.
+    */
+  def constructFromString(resultType: Class[_], unitType: Class[_], value: String*): Try[Any] = Try {
+    if (resultType != unitType && !ReflectionUtil.isCollectionClass(resultType) && !classOf[Option[_]].isAssignableFrom(resultType)) {
+      throw new ReflectionUtilException("Don't know how to make a " + resultType.getSimpleName)
+    }
+
+    try {
+      typedValueFromString(resultType, unitType, value:_*).get
+    }
+    catch {
+      case e: NoSuchMethodException =>
+        throw new ReflectionUtilException(s"Cannot find string ctor for '${unitType.getSimpleName}'", e)
+      case e: InstantiationException =>
+        throw new ReflectionUtilException(s"Abstract class '${unitType.getSimpleName} cannot be used for an argument value type.", e)
+      case e: IllegalAccessException =>
+        throw new ReflectionUtilException(s"String constructor for argument class ''${unitType.getSimpleName}'' must be public.", e)
+      case e: InvocationTargetException =>
+        throw new ReflectionUtilException(
+          s"Problem constructing '${unitType.getSimpleName}' from the string${plural(value.size)} '" +
+            value.toList.mkString(", ") + "'."
+        )
+    }
+  }
+
+  /**
+    * Attempts to construct one or more unitType values from Strings, and return them as the resultType. Handles
+    * the creation of and packaging into collections as necessary.
+    */
+  def typedValueFromString(resultType: Class[_], unitType: Class[_], value: String*): Try[Any] = Try {
```

Comment:
Should this be private, since it's basically the guarded implementation of `constructFromString`?